libtsuba/src/tsuba.cpp

```diff
@@ -232,6 +233,158 @@ tsuba::ListAvailableViews(const std::string& rdg_dir) {
+  auto rdg_views =
+      KATANA_CHECKED(tsuba::ListAvailableViewsFromVersion(src_dir, version));
+  for (const auto& rdg_view : rdg_views) {
+    KATANA_LOG_WARN("view_path: {}", rdg_view.view_path);
+    auto uri = KATANA_CHECKED(katana::Uri::Make(src_dir));
+
+    auto rdg_manifest_res =
+        tsuba::RDGManifest::Make(uri, rdg_view.view_type, version);
+    if (!rdg_manifest_res) {
+      continue;
+    }
+
+    auto fnames = KATANA_CHECKED(rdg_manifest_res.value().FileNames());
+    for (auto fname : fnames) {
+      auto src_file_path = katana::Uri::JoinPath(src_dir, fname);
+      auto src_file_uri = KATANA_CHECKED(katana::Uri::Make(src_file_path));
+
+      // If we are a manifest, we need to change our version to 1.
+      if (tsuba::RDGManifest::IsManifestUri(src_file_uri)) {
+        continue;
+      }
+
+      // Check to see if we have a partition file
+      // If we have a partition file, the dst path should be based on PartitionFileName using rdg manifest info
+      // We're batching this now because we want to rely on having the RDG manifest file in case things
+      // change in the future.
+      katana::Uri dst_file_uri;
+      if (tsuba::RDGManifest::IsPartitionFileUri(src_file_uri)) {
+        KATANA_LOG_WARN("src_file_uri partition: {}", src_file_uri);
+        auto host_id =
+            KATANA_CHECKED(tsuba::RDGManifest::ParseHostFromPartitionFile(
+                src_file_uri.BaseName()));
+        auto dst_dir_uri = KATANA_CHECKED(katana::Uri::Make(dst_dir));
+        dst_file_uri = tsuba::RDGManifest::PartitionFileName(
+            rdg_manifest_res.value().view_type(), dst_dir_uri, host_id, 1);
+        KATANA_LOG_WARN("dst_file_uri partition: {}", dst_file_uri);
+      } else {
+        auto dst_file_path = katana::Uri::JoinPath(dst_dir, fname);
+        dst_file_uri = KATANA_CHECKED(katana::Uri::Make(dst_file_path));
+        KATANA_LOG_WARN("src_file_uri: {}", src_file_uri);
+        KATANA_LOG_WARN("dst_file_uri: {}", dst_file_uri);
+      }
+
+      src_dst_pairs.push_back(std::make_pair(src_file_uri, dst_file_uri));
+    }
+
+    // We add the manifest file to the vector
+    // Set the version to be 1
+    auto rdg_manifest_uri = rdg_manifest_res.value().FileName();
+    rdg_manifest_res.value().ResetVersion();
+    auto dst_rdg_manifest_path = katana::Uri::JoinPath(
+        dst_dir, rdg_manifest_res.value().FileName().BaseName());
+    auto dst_rdg_manifest_uri =
+        KATANA_CHECKED(katana::Uri::Make(dst_rdg_manifest_path));
+    KATANA_LOG_WARN("rdg_manifest_uri: {}", rdg_manifest_uri);
+    KATANA_LOG_WARN("dst_rdg_manifest_uri: {}", dst_rdg_manifest_uri);
+    src_dst_pairs.push_back(
+        std::make_pair(rdg_manifest_uri, dst_rdg_manifest_uri));
+  }
+  return src_dst_pairs;
+}
+
+katana::Result<void>
+tsuba::CopyRDG(std::vector<std::pair<katana::Uri, katana::Uri>> src_dst_pairs) {
+  // TODO: make sure that manifests are written at the end!
+  // TODO: add do_all loop
```

Comment:
`TODO`s should have a name (probably yours) attached. Useful because sometimes things get moved and finding the author through blame is complicated. For instance: `TODO(thunt)`